truss/templates/shared/lazy_data_resolver.py

```diff
@@ -38,28 +43,68 @@ class BasetenPointerManifest(pydantic.BaseModel):
 class LazyDataResolver:
     def __init__(self, data_dir: Path):
         self._data_dir: Path = data_dir
-        self._bptr_resolution: Dict[str, str] = _read_bptr_resolution()
+        self._bptr_resolution: Dict[str, Tuple[str, str]] = _read_bptr_resolution()
         self._resolution_done = False
+        self._uses_b10_cache = eval(os.environ.get(BASETEN_FS_ENABLED_ENV_VAR, "False"))
+
+    def cached_download_from_url_using_requests(
+        self, URL: str, hash: str, file_name: str
+    ):
+        """Download object from URL, attempt to write to cache and symlink to data directory if applicable, data directory otherwise.
+        In case of failure, write to data directory
+        """
+        if self._uses_b10_cache:
+            file_path = CACHE_DIR / hash
+            if file_path.exists():
+                os.symlink(file_path, self._data_dir / file_name)
```

Comment:
There's no scenario where the symlink will already exist in the data dir, right? Because this will only be called the first time the container is starting? os.symlink will throw an error if it already exists.

Reply:
Yup this only happens at startup so should be safe to assume that there aren't symlinks here